### PR TITLE
Patch build.go so metadata works with "fyne package" when building app based on fyne@develop

### PR DESCRIPTION
**Pull request by roffe**
Patch build.go so metadata works when "fyne package"ing app built on latest develop branch of fyne

### CODE REVIEW

**Comment by andydotxyz** [comment]
Please provide more info on how you can trigger this problem - to our knowledge the build is not broken with develop branch.

**Comment by roffe** [comment]
```
go install fyne.io/tools/cmd/fyne@latest
```
then inside project
```
go get fyne.io/fyne/v2@develop
```
then
```
fyne package --release
```
no fyne_metadata_init.go is created and the app when started will have metadata version 0.0.1 and build no 1

I'm packaging my app right now with this patched version because I kept swearing for a week that develop branch broke again when packaging apps and then decided to do something about it.

**Comment by andydotxyz** [comment]
What version of Go are you using?

**Comment by roffe** [comment]
go version go1.22.0 windows/amd64

**Comment by roffe** [comment]
here is some debug output i had put in while figuring out why the cli wasn't generating the metadata go file.
```
2024/03/31 14:49:05 injectMetadataIfPossible
2024/03/31 14:49:05 fyneGoModVersion v2.4.4-rc2.0.20240329173153-506db546dbfd
2024/03/31 14:49:05 Creating metadata init file
2024/03/31 14:49:11 injectMetadataIfPossible
2024/03/31 14:49:11 fyneGoModVersion v2.4.4-rc2.0.20240329173153-506db546dbfd
2024/03/31 14:49:11 Creating metadata init file
```

**Comment by andydotxyz** [comment]
> It looks like maybe Go 1.22 changed how versions of unreleased software are labelled...

Or perhaps it was because you downloaded it during a release cycle for the 2.4.x branch.
I suspect that a "go mod tidy" right now would fix the issue as no release candidate is under way.

**Comment by andydotxyz** [comment]
> > It looks like maybe Go 1.22 changed how versions of unreleased software are labelled...
> 
> Or perhaps it was because you downloaded it during a release cycle for the 2.4.x branch. I suspect that a "go mod tidy" right now would fix the issue as no release candidate is under way.

It must be the latter - Go 1.22 testing here shows no change

**Comment by roffe** [comment]
can confirm problem still exists even after go mod tidy and downloading latest develop and reinstalling the fyne cmd

```
go mod tidy
go install fyne.io/tools/cmd/fyne@latest
go get fyne.io/fyne/v2@develop          
go: upgraded fyne.io/fyne/v2 v2.4.4-rc2.0.20240329173153-506db546dbfd => v2.4.4-rc2.0.20240401200252-f528f23733b7
fyne package --release
```

**Comment by andydotxyz** [comment]
Thanks for this info. I can also replicate it now. Somehow the version depends on state of other branches which I don't understand but the patch does resolve it!

**Review by andydotxyz — APPROVED** [review]

File: cmd/fyne/internal/commands/build.go [hunk 1]
```diff
@@ -443,5 +443,9 @@ func normaliseVersion(str string) string {
 	if pos := strings.Index(str, "-0.20"); pos != -1 {
 		str = str[:pos] + "-dev"
 	}
+
```
Comment (andydotxyz) [inline cmd/fyne/internal/commands/build.go:446]: Looks like there is a trailing tab character in here

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Failing static check because of extra space

File: cmd/fyne/internal/commands/build.go (hunk 1 above)
Comment (roffe) [reply cmd/fyne/internal/commands/build.go]: fixed

**Review by andydotxyz — APPROVED** [review]
Thanks

**Pull request merged**